include/pagmo/algorithms/pso.hpp

```diff
@@ -254,9 +267,10 @@ class PAGMO_DLL_PUBLIC pso
     unsigned m_neighb_param;
     // memory
     bool m_memory;
+    mutable std::optional<pso::memory> m_memory_data;
```

Comment:
``std::optional`` might be problematic because I don't think it is supported by the Boost.serialization library. If this is indeed the case, replacing it with ``boost::optional`` should be straightforward.